Better docs for crypto operations (#323)
Talus-Network/nexus-sdk: docs/guides/setup.md

## Configure Encryption for Nexus workflows

> Nexus encrypts every sensitive value in your CLI config and every DAG payload using a Signal-inspired stack: a persistent master key protects secrets at rest, an X3DH identity key authenticates you to the network, and a Double-Ratchet session derived from an on-chain pre-key encrypts runtime traffic. Follow the steps below in order; each one builds on the previous.

Author: [7a8c897](https://github.com/Talus-Network/nexus-sdk/pull/323/commits/7a8c8973dac416e799baeb12923dc90eb862d32b)

### 1. Initialize the CLI master key

The CLI stores encrypted blobs (identity key, sessions, Walrus credentials, etc.) in `~/.nexus/*.toml`. Those blobs are decrypted using a 32-byte master key that lives either in your OS keyring or is derived from a passphrase via Argon2id.

```bash
# Option A: generate a raw master key inside the OS keyring
nexus crypto init-key

# Option B: store a passphrase (useful for headless or CI)
printf "my-strong-passphrase" | nexus crypto set-passphrase --stdin
```

Both commands refuse to overwrite existing credentials unless you add `--force`, because rotating the master key invalidates every encrypted entry. You can confirm which source will be used by running:

```bash
nexus crypto key-status
```

### 2. Generate an identity key

> Your long-term identity key is the “public face” of the CLI in the Signal/X3DH handshake. It only needs to be generated once per installation—or whenever you intentionally rotate it (which also invalidates stored sessions).

Reviewer: ```suggestion
Your long-term identity key represents the "public face" of the CLI in the Signal/X3DH handshake. It only needs to be generated once per installation or when you intentionally rotate it, which also invalidates stored sessions.
```

```bash
nexus crypto generate-identity-key
```

This writes a freshly generated X25519 key pair into `~/.nexus/crypto.toml`, encrypted with the master key from Step 1.

### 3. Establish a Signal-style session

With persistence in place, run the auth flow to claim a pre-key bundle on-chain and derive a per-session Double Ratchet. The CLI handles the entire exchange for you:

```bash
nexus crypto auth
```

Behind the scenes the command:

1. Submits a programmable transaction that calls `pre_key_vault::claim_pre_key`, producing a pre-key bundle owned by your Sui address.

> 1. Runs the X3DH sender flow locally using your identity key and that bundle, deriving shared secrets and the first Double-Ratchet message.

Reviewer: > Runs the X3DH sender flow locally

Do we have any sort of chart/diagram to visualize what's happening behind the scenes? That would be helpful for us as well.
